internal/testing/enginetest/enginetest.go

```diff
@@ -326,55 +334,75 @@ func RunTestEngine_NewModuleEngine_InitTable(t *testing.T, et EngineTester) {
 	})
 }
 
-func runTestModuleEngine_Call_HostFn_ModuleContext(t *testing.T, et EngineTester) {
-	features := wasm.Features20191205
-	e := et.NewEngine(features)
-
-	sig := &wasm.FunctionType{
-		Params:           []wasm.ValueType{wasm.ValueTypeI64},
-		Results:          []wasm.ValueType{wasm.ValueTypeI64},
-		ParamNumInUint64: 1, ResultNumInUint64: 1,
-	}
+func runTestModuleEngine_Call_HostFn_Mem(t *testing.T, et EngineTester, readMem *wasm.Code) {
+	e := et.NewEngine(wasm.Features20191205)
+	host, importing, close := setupCallMemTests(t, e, readMem, et.ListenerFactory())
+	defer close()
 
-	memory := &wasm.MemoryInstance{}
-	var mMemory api.Memory
-	host := reflect.ValueOf(func(m api.Module, v uint64) uint64 {
-		mMemory = m.Memory()
-		return v
-	})
+	hostMemoryVal := uint64(3)
+	host.Memory = &wasm.MemoryInstance{Buffer: u64.LeBytes(hostMemoryVal), Min: 1, Cap: 1, Max: 1}
+	importingMemoryVal := uint64(6)
+	importing.Memory = &wasm.MemoryInstance{Buffer: u64.LeBytes(importingMemoryVal), Min: 1, Cap: 1, Max: 1}
 
-	m := &wasm.Module{
-		TypeSection:     []*wasm.FunctionType{sig},
-		FunctionSection: []wasm.Index{0},
-		CodeSection:     []*wasm.Code{{GoFunc: &host}},
+	tests := []struct {
```

Comment:
rewrote this test which only covered one case, to cover direct and indirect host function calls that access memory. notably, this includes a host wasm function which calls another host wasm function, and proves the nested call uses the caller memory, not the host.